microsoft-edge/devtools-guide-chromium/whats-new/2024/03/devtools-123.md

```diff
@@ -0,0 +1,117 @@
+---
+title: What's New in DevTools (Microsoft Edge 123)
+description: "Memory tool reports backing store sizes as part of an object's shallow size. Crash analyzer tool is available by default. Command Palette experimental feature has been removed. Undocked DevTools window is kept focused as the topmost window while inspecting a page. And more."
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.service: microsoft-edge
+ms.subservice: devtools
+ms.date: 03/21/2024
+---
+# What's New in DevTools (Microsoft Edge 123)
+
+[!INCLUDE [Microsoft Edge team note for top of What's New](../../includes/edge-whats-new-note.md)]
+
+
+<!-- ====================================================================== -->
+## Memory tool reports backing store sizes as part of an object's shallow size
+
+<!-- Subtitle: New experiment is available to make Shallow Size column in the Memory tool more helpful. -->
+
+Microsoft Edge 123 adds a new experiment called **In heap snapshots, treat backing store size as part of the containing object**.  If you enable this experiment, the **Memory** tool reports shallow sizes in a way that more closely matches JavaScript behavior.  By enabling this experiment, containers and their contents can now be represented as a single item, such as:
+* An Object and its properties.
+* An Array and its contents.
+* A Map and its contents.
+* A Set and its contents.
+
+For example, to find the largest arrays, you can sort Array​ instances by shallow size, as shown below.
+
+Default behavior for heap snapshots:
+![Heap snapshot default behavior](./devtools-123-images/heap-snapshot-default.png)
+
+Heap snapshots with the new experiment enabled:
+![Heap snapshot with experiment enabled](./devtools-123-images/heap-snapshot-with-experiment.png)
+
+This experiment is helpful because heap snapshots in the Memory tool show a complete list of everything allocated in the V8 and Blink heaps.  Sometimes that list is a little too fine-grained, breaking out an item as if it is multiple items.  For example, the list in the heap snapshot represents each JavaScript Array instance separately from the contents of that Array, which are depicted as _object elements_.<!--todo: link-->  However, in JavaScript code, an Array is a single entity, rather than an array instance and separate contents of the array.
```

Comment:
link object elements
@leahmsft

Reply:
sorry i didn't make it clear that `(object elements)[]` is the name of the object. will fix in next commit